services/src/main/java/org/keycloak/locale/DefaultLocaleSelectorProvider.java

```diff
@@ -61,7 +68,7 @@ public Locale resolveLocale(RealmModel realm, UserModel user) {
         return Locale.ENGLISH;
     }
 
-    private Locale getUserLocale(RealmModel realm, AuthenticationSessionModel session, UserModel user, HttpHeaders requestHeaders) {
+    private Locale getUserLocale(RealmModel realm, AuthenticationSessionModel session, UserModel user, HttpHeaders requestHeaders, Theme.Type themeType) {
```

Comment:
I find the implementation of `getUserLocale()` very confusing, it has a tonne of fallback cases that seem like they might not be relevant at all for the case of an administrative user performing an action on behalf of another user whom is the recipient of said action (e.g. sending an e-mail).

For example, `getLocaleCookieSelection()` is called further down, which in the case of the e-mail theme also seems incorrect to use as a source of truth, since it would reflect the `LOCALE` cookie of the currently authenticated user, not the user that is acted upon which will receive the e-mail.

I would expect a special code path here for 'If a user acts on behalf another user', which could be the one of the first conditions. From what I gather here the `user` passed into this method can be different than the user that is authenticated in the `session`, so that could be used to reliably determine this code path.

For example:

```java
private Locale getUserLocale(RealmModel realm, AuthenticationSessionModel session, UserModel user, HttpHeaders requestHeaders) {
    // If the authenticated user is acting on another users' behalf the only way
    // to determine the preferred locale is to retrieve it from the users' preferences.
    if (!session.getAuthenticatedUser().getId().equals(user.getId())) {
        return getUserProfileSelection(realm, user);
    }
    
    // Rest of the regular locale detection code here.
}
```

This would also remove the need to pass the `themeType` around and modify existing interfaces.

Reply:
@jonkoops Thanks for the review! I see your point and agree that `getUserLocale()` is too generic, and some controls are unnecessary in certain situations. Indeed, my first approach was similar to the one you pointed out https://github.com/graziang/keycloak/commit/7d79803c1d2c7c16ff1a3b6b32c015616c043a61: ignoring the `Accept-Language` header when the logged user is an admin and is different from the user in the parameter of `resolveLocale()`. However, this approach felt a bit forced, as `session.getAuthenticatedUser()`, when set, has the same user of the user in the parameter and in the case of sending the email with an execute-actions-email admin api request, `session.getAuthenticatedUser()` is null and I haven't found a clean solution.


Another reason why IMO it may make sense to always ignore `Accept-Language` header when you have the email theme is when a user is not logged, for instance, when a user tries to reset password from the login screen.
In this case, the user in `session.getAuthenticatedUser()` would be the same as the user in the parameter of `resolveLocale()`, and if the user does not have the locale, it is possible to send them an email in any allowed language in the realm. WDYT? Adding @mposolda

Reply:
@graziang @jonkoops Question 1 is, what we want to achieve regarding the behaviour? To me, it makes sense to do something as proposed exactly in the GH issue and in the comment on it: https://github.com/keycloak/keycloak/issues/10233 . If we do this, it means that for `email` theme, we will ignore `Accept-Language` header and the Locale cookie as those are set in the browser and not in the email client. If we do this always for `email` theme, we don't need to care about comparing users from authentication session etc. Does this makes sense to you?

Question 2 implementation: Regarding implementation, I would also rather do something like:

```
if (theme.type == email) {
  user -> realm default -> EN
} else {
   // All as done today
}
```
instead of differ between email and other types at individual implementation methods. WDYT?

Reply:
My vote is to always ignore the `Accept-Language` header for email themes instead of comparing users. This approach avoids the need for additional logic to verify if the user is the same and if is admin, which is not always available in `session.getAuthenticatedUser()` and risks further complicating the methods. I also double-checked that the locale for a user is not set after registration, even with internationalization enabled. There is a related issue: #13505.

I moved the theme check up in `getUserLocale()` to skip unnecessary checks. Can it work?

Reply:
Regarding #13505 this might be fixed via #32687.